src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,30 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+interface Props {
+  todo: Todo | undefined,
+  closeModal: () => void
+}
+
+export const TodoModal: React.FC<Props> = ({ todo, closeModal }) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    // eslint-disable-next-line max-len
+    fetch(`https://mate-academy.github.io/react_dynamic-list-of-todos/api/users/${todo?.userId}.json`)
+      .then(response => response.json())
+      .then(userFromServer => {
+        setUser(userFromServer);
+      });
+  }, [user]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {!user ? (
```

Comment:
instead of checking for !user you should check for loading

add `const [loading, setLoading] = useState(false);` and set `setLoading(true)` in useEffect